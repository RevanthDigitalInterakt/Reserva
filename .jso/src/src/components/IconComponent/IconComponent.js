  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _objectWithoutProperties2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[2]));
  var _reactNativeAutoHeightImage = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[3]));
  var _icons = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[4]));
  var _excluded = ["icon"];
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  var defaultIconSize = 60;
  function IconComponent(_ref) {
    var icon = _ref.icon,
      rest = (0, _objectWithoutProperties2.default)(_ref, _excluded);
    if (!icon) {
      throw new Error('Required icon name');
    }
    if (Object.keys(_icons.svgs).includes(icon)) {
      var ComponentSvg = _icons.default[icon];
      return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[5]).jsx)(ComponentSvg, {});
    }
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[5]).jsx)(_reactNativeAutoHeightImage.default, Object.assign({
      resizeMode: "contain",
      source: _icons.default[icon],
      width: defaultIconSize,
      height: defaultIconSize
    }, rest));
  }
  var _default = exports.default = (0, _react.memo)(IconComponent);
