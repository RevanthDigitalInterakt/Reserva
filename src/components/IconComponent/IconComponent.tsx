import React, { memo } from 'react';
import AutoHeightImage from 'react-native-auto-height-image';
import icons, { TIcons, svgs } from '../../base/styles/icons';

export interface TSource {
  uri: string;
}

interface IProps extends Omit<React.ComponentProps<typeof AutoHeightImage>, 'source' | 'width' | 'height'> {
  icon: TIcons,
  source?: TSource | number
  width?: number;
  height?: number;
}

const defaultIconSize = 60;

function IconComponent({ icon, ...rest }: IProps) {
  if (!icon) {
    throw new Error('Required icon name');
  }

  if (Object.keys(svgs).includes(icon)) {
    const ComponentSvg = icons[icon];
    return <ComponentSvg />;
  }

  return (
    <AutoHeightImage
      resizeMode="contain"
      source={icons[icon]}
      width={defaultIconSize}
      height={defaultIconSize}
      {...rest}
    />
  );
}

export default memo(IconComponent);
