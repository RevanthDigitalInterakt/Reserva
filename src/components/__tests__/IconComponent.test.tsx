import React from 'react';
import {
  render, waitFor,
} from '@testing-library/react-native';
import IconComponent from '../IconComponent/IconComponent';

describe('IconComponent', () => {
  it('should match with the snapshot', async () => {
    const Component = (
      <IconComponent
        icon="whatsapp"
      />
    );

    const { toJSON } = await waitFor(() => render(Component));

    expect(toJSON()).toMatchSnapshot();
  });
});
