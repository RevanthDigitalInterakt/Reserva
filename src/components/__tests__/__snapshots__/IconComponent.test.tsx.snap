// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconComponent should match with the snapshot 1`] = `
<Image
  height={60}
  onError={[Function]}
  resizeMode="contain"
  source={
    Object {
      "commons": Object {
        "GuideBermuda": [Circular],
        "GuideCalca": [Circular],
        "GuideCamiseta": [Circular],
        "GuideHoodie": [Circular],
        "GuideJaqueta": [Circular],
        "GuideMangaCurta": [Circular],
        "GuideMangaLonga": [Circular],
        "GuidePolo": [Circular],
        "GuideShort": [Circular],
        "GuideSueter": [Circular],
        "GuideSungaCueca": [Circular],
        "QrcodeBackground": [Circular],
        "americanExpress": [Circular],
        "arrowInstruction": [Circular],
        "bagEmpty": [Circular],
        "bannerCatalog": [Circular],
        "bannerLogin": [Circular],
        "bannerOffer": [Circular],
        "cardFront": [Circular],
        "cardVerso": [Circular],
        "clothingCare1": [Circular],
        "clothingCareBleach": [Circular],
        "clothingCareDrying": [Circular],
        "clothingCarePass": [Circular],
        "clothingCareProfessionalWashing": [Circular],
        "clothingCareWash": [Circular],
        "cupomModalBackground": [Circular],
        "dividerReserva": [Circular],
        "elo": [Circular],
        "emptyProductCatalog": [Circular],
        "facebook": [Circular],
        "foraDoAr": [Circular],
        "headerLogin": [Circular],
        "homeMock1": [Circular],
        "homeMock2": [Circular],
        "homeMock3": [Circular],
        "homeMock4": [Circular],
        "imageCashback": [Circular],
        "imageNotFound": [Circular],
        "infoFill": [Circular],
        "instagram": [Circular],
        "localReserva": [Circular],
        "logoQRCode": [Circular],
        "mastercard": [Circular],
        "noStoresFound": [Circular],
        "noWishList": [Circular],
        "pinYou": [Circular],
        "purchaseConfirmation": [Circular],
        "raceImageBackground": [Circular],
        "saleOff": [Circular],
        "searchNotFound": [Circular],
        "selectRectangle": [Circular],
        "shirt1": [Circular],
        "shirt2": [Circular],
        "shirt3": [Circular],
        "shirt4": [Circular],
        "topBannerMock1": [Circular],
        "twitter": [Circular],
        "update": [Circular],
        "visa": [Circular],
        "whatsapp": [Circular],
        "withoutInternet": [Circular],
        "zoomHand": [Circular],
      },
      "default": Object {
        "GuideBermuda": [Circular],
        "GuideCalca": [Circular],
        "GuideCamiseta": [Circular],
        "GuideHoodie": [Circular],
        "GuideJaqueta": [Circular],
        "GuideMangaCurta": [Circular],
        "GuideMangaLonga": [Circular],
        "GuidePolo": [Circular],
        "GuideShort": [Circular],
        "GuideSueter": [Circular],
        "GuideSungaCueca": [Circular],
        "QrcodeBackground": [Circular],
        "americanExpress": [Circular],
        "arrowInstruction": [Circular],
        "bagEmpty": [Circular],
        "bannerCatalog": [Circular],
        "bannerLogin": [Circular],
        "bannerOffer": [Circular],
        "cardFront": [Circular],
        "cardVerso": [Circular],
        "chevronRight": [Function],
        "clothingCare1": [Circular],
        "clothingCareBleach": [Circular],
        "clothingCareDrying": [Circular],
        "clothingCarePass": [Circular],
        "clothingCareProfessionalWashing": [Circular],
        "clothingCareWash": [Circular],
        "cupomModalBackground": [Circular],
        "dividerReserva": [Circular],
        "elo": [Circular],
        "emptyProductCatalog": [Circular],
        "facebook": [Circular],
        "foraDoAr": [Circular],
        "headerLogin": [Circular],
        "homeMock1": [Circular],
        "homeMock2": [Circular],
        "homeMock3": [Circular],
        "homeMock4": [Circular],
        "imageCashback": [Circular],
        "imageNotFound": [Circular],
        "infoFill": [Circular],
        "instagram": [Circular],
        "localReserva": [Circular],
        "logoPrime": [Function],
        "logoQRCode": [Circular],
        "mastercard": [Circular],
        "noStoresFound": [Circular],
        "noWishList": [Circular],
        "pinYou": [Circular],
        "purchaseConfirmation": [Circular],
        "raceImageBackground": [Circular],
        "saleOff": [Circular],
        "searchNotFound": [Circular],
        "selectRectangle": [Circular],
        "shirt1": [Circular],
        "shirt2": [Circular],
        "shirt3": [Circular],
        "shirt4": [Circular],
        "topBannerMock1": [Circular],
        "twitter": [Circular],
        "update": [Circular],
        "visa": [Circular],
        "whatsapp": [Circular],
        "withoutInternet": [Circular],
        "zoomHand": [Circular],
      },
      "svgs": Object {
        "chevronRight": [Function],
        "logoPrime": [Function],
      },
    }
  }
  style={
    Array [
      Object {
        "height": 0,
        "width": 60,
      },
      undefined,
    ]
  }
/>
`;
